CacheMeIfYouCan/CacheMeIfYouCan/CacheTypes/FileSystemDataCache.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  FilesystemDataCache.swift
+//  CacheMeIfYouCan
+//
+//  Created by Ellen Shapiro on 12/10/18.
+//  Copyright © 2018 Bakken & Baeck. All rights reserved.
+//
+
+import Foundation
+
+open class FileSystemDataCache<T: DataConvertible>: Cache {
+    
+    public typealias StoredType = T
+    
+    let folderURL: URL
+    
+    public lazy var localQueue = FileSystemDataCache.defaultQueue
+    
+    public init(folderURL: URL) {
+        self.folderURL = folderURL
+        
+        do {
+            try FileManagerHelper.createFolderIfNeeded(at: folderURL)
+        } catch {
+            LogHelper.log("Error creating folder: \(error)")
+        }
+    }
+    
+    private func localURL(for url: URL) -> URL {
+        let fileName = FilesystemPathHelper.fileName(from: url)
+        return self.folderURL
+            .appendingPathComponent(fileName)
+    }
+    
+    open func actuallyStore(item: T, for url: URL) throws {
+        guard let data = item.toData else {
+            return
+        }
+        
+        let url = self.localURL(for: url)
+        try data.write(to: url)
+    }
+    
+    open func actuallyFetchItem(for url: URL) throws -> T? {
+        let url = self.localURL(for: url)
+        guard FileManagerHelper.fileExists(at: url) else {
+            return nil
+        }
+    
+        let data = try Data(contentsOf: url)
+        let item = T(data: data)
+
+        return item
+    }
+    
+    open func clearAll() throws {
+        try FileManagerHelper.removeContentsOfFolder(at: self.folderURL)
+    }
+
+    open func actuallyRemoveItem(for url: URL) throws {
+        let localURL = self.localURL(for: url)
+        try FileManagerHelper.removeFile(at: localURL)
+    }
+    
+    /// Gets the item from the cache, or if it's not present in the cache, tries to download it from the given URL
```

Comment:
Just a note to make sure that this implementation detail is extremely important. This lib doesn't do any kind of HTTP Caching (which external users of this library may be expecting), so best to make this very clear in the README.